app/controllers/register_controller.rb

```diff
@@ -63,20 +63,50 @@ def create
 
     payload = generate_canonical_object(fields, params)
 
-    @change = Change.new(register_name: params[:register], payload: payload, user_id: current_user.id)
-    @change.status = Status.new(status: 'pending')
-    @change.save
+    if current_user.basic?
+      @change = Change.new(register_name: params[:register], payload: payload, user_id: current_user.id)
+      @change.status = Status.new(status: 'pending')
+      @change.save
 
-    @change_approvers = Register.find_by(key: params[:register]).team.team_members.where.not(role: 'basic', user_id: current_user)
+      @change_approvers = Register.find_by(key: params[:register]).team.team_members.where.not(role: 'basic', user_id: current_user)
 
-    if @change_approvers.present?
-      RegisterUpdatesMailer.register_update_notification(@change, current_user, @change_approvers).deliver_now
-    end
+      if @change_approvers.present?
+        RegisterUpdatesMailer.register_update_notification(@change, current_user, @change_approvers).deliver_now
+      end
 
-    RegisterUpdatesMailer.register_update_receipt(@change, current_user).deliver_now
+      RegisterUpdatesMailer.register_update_receipt(@change, current_user).deliver_now
 
-    flash[:notice] = 'Your update has been submitted, you\'ll recieve a confirmation email once the update is live'
-    redirect_to action: 'index', register: params[:register]
+      flash[:notice] = 'Your update has been submitted, you\'ll recieve a confirmation email once the update is live'
+      redirect_to action: 'index', register: params[:register]
+    else
+      if params[:confirm_approve] == '1'
+        @change = Change.new(register_name: params[:register], payload: payload, user_id: current_user.id)
+        @change.status = Status.new(status: 'approved', reviewed_by_id: current_user.id)
+        @change.save
+
+        rsf_body = CreateRsf.(@change.payload, @change.register_name)
+
+        Rails.env.development? || Rails.env.staging? || response = RegisterPost.(@change.register_name, rsf_body)
```

Comment:
Since this is repeated could we move it to a variable e.g. `is_non_prod = Rails.env.development? || Rails.env.staging?`

Reply:
I feel like this whole block is wrong anyway i.e. we should only run this if prod and not check for all the other environments

Reply:
yeah, possibly although I think the end to end tests in env `test` expect the POST to happen then mock the response, which is a useful part of the test. Maybe we could make this a per env configuration value like `post_to_register`?

Reply:
You are right, our tests would fail but I dont think thats a reason to not refactor... maybe another story :)